### DISCUSSION TITLE
ERR_CERTIFICATE_TRANSPARENCY_REQUIRED

### DISCUSSION THREAD

**Discussion by emilypeto** [mitmproxy/mitmproxy#5720] [Intercept Everything!]
Edit: this issue and the solution are discussed at length here: https://httptoolkit.com/blog/chrome-android-certificate-transparency/

Steps taken:
- Install mitmproxy on Windows 11 using the windows installer from the mitmproxy website
- Run mitmproxy
- From Android 12 device, configure proxy and visit mitm.it and download the certificate
- Install the certificate from settings (Install CA certificate)
- Utilize magisk module to move the certificate to the trusted store
- Configure proxy and visit any site over TLS
- Chrome v106 presents the untrusted certificate warning: ERR_CERTIFICATE_TRANSPARENCY_REQUIRED

Also note that when attempting to install the mitmproxy certificate for "VPN and Apps" instead, it yields the error "This file can't be used as a VPN or app certificate."

**Comment by mhils** [comment]
The problem here is that Chrome requires Certificate Transparency log entries for all certificates that that trace back to system CAs. If you add mitmproxy as a user-added CA (not via magsik, but manual install), this error should go away.

I'm not sure what happens if a CA is both present as a system CA and user-added CA. Does that still work? Could you take a look and report back? ;)

**Reply by emilypeto** [reply]
When installed as a user certificate (install CA certificate), I get ERR_CERT_AUTHORITY_INVALID. Android 12 will not trust any certificates that are not in the trusted store.

Installation as both trusted and user yields the same ERR_CERTIFICATE_TRANSPARENCY_REQUIRED issue.

**Reply by mhils** [reply]
As far as I know Chrome opts in to accept user-added CAs. It's different for non-browser apps of course.

**Reply by emilypeto** [reply]
I wanted to use mitmproxy for other apps, which are running into the same untrusted certificate issues. Is this just a limitation of mitmproxy with respect to Android 12 then?

Edit: not sure what I did, but it's working now for apps other than Chrome.

**Reply by mhils** [reply]
You want system store for most apps, user store for Chrome. I'm not sure what happens if you install the same CA in both (that was my earlier question). Does that make sense?

**Reply by lexnel** [reply]
If you install the CA certificate in both trust stores, Chrome uses the one in the system store and therefore throws the same error as mentioned above.

**Comment by nikicat** [comment]
Hello, what is a recommended solution to intercept all web requests from Android (including Chrome and other apps)?

**Reply by lexnel** [reply]
It depends. If you want to intercept the traffic from Chrome, it is sufficient to put your mitmproxy certificate in the clients certificate trust store. However if you want to intercept general app traffic, you need to add it to the system trust store. You could use something like [that](https://github.com/Magisk-Modules-Alt-Repo/custom-certificate-authorities).

**Comment by nutterthanos** [comment]
I get this error with chrome or anything using any google chrome sorta webview

NET::ERR_CERT_AUTHORITY_INVALID

Android 10, Magisk Module Installed and added to user certificates

**Comment by TwoNull** [comment]
I also had this issue while using Magisk to add the Mitmproxy cert to the system store. After trying all of the solutions previously discussed in this thread, I opted to just install an older build of Chromium (v98 or older should work fine).

**Comment by LyraHarring** [comment]
The error ['ERR_CERTIFICATE_TRANSPARENCY_REQUIRED](https://certera.com/kb/how-to-fix-the-neterr_certificate_transparency_required-error/) is a message commonly encountered in Google Chrome. It signifies a discrepancy in the website’s SSL cert, mainly related to Certificate Transparency. It is mainly caused by issues related to website SSL certificates. There are n numbers of reasons for the error to have occurred, which are as follows:-

1) Certificate not added to transparency log
2)  Incorrectly installed SSL certificate
3) Corrupted SSL Certificate
4)  Expired SSL Certificate

There are many different methods for solving the error, which you need to understand properly in detail. Hope it helps!
